### PR TITLE
attachemnts and row size tests

### PR DESCRIPTION
**Pull request by snopoke**

### CODE REVIEW

**Comment by czue** [comment]
looks good. is it clear we want to store the attachments in the DB? I thought we might consider just storing pointers to cold storage or something instead

File: prototype/tests/test_models.py [hunk 1]
```diff
@@ -67,3 +73,142 @@ def test_case_versioning(self, testapp):
...
+                _random_string(20): {
+                    'mime': random.choice(ATTACHMENT_TYPES.values()),
+                    'size': 5000,
+                    'id': _random_string(20),
+                    'src': _random_string(50),
+                    'from': 'local',
+                    'name': _random_string(20),
+                    'md5': hashlib.md5(str(uuid4())).hexdigest(),
+                    'props': {'w': 240, 'h': 164}
+                }
```
Comment (czue) [inline prototype/tests/test_models.py:137]: we should reduce some of these properties if they're not part of the spec

File: prototype/tests/test_models.py [hunk 2]
```diff
@@ -67,3 +73,142 @@ def test_case_versioning(self, testapp):
...
+            ...
+        },
+        'p': {
+            'property_name': [form_id_prefix, value]
+            ...
+        }
+    }
+
+    The 'unique_form_id_prefix' is as many characters as are needed from the front of the form_id
+    in order to make it unique in the 'f' dict.
```
Comment (czue) [inline prototype/tests/test_models.py:192]: this is clever.

Comment (czue) [inline prototype/tests/test_models.py:192]: i'm still not 100% sure we need to store any of this if we only support rebuildng cases from forms (and not from themselves)

File: prototype/tests/test_models.py (hunk 1 above)
Comment (snopoke) [inline prototype/tests/test_models.py:137]: what spec do you mean?

Comment (czue) [inline prototype/tests/test_models.py:137]: https://github.com/dimagi/commcare/wiki/CaseAttachmentAPI

**Pull request merged**